src/Microsoft.Diagnostics.Monitoring.EventPipe/DiagnosticsEventPipeProcessor.cs

```diff
@@ -69,21 +81,33 @@ public async Task Process(DiagnosticsClient client, int pid, TimeSpan duration,
                     {
                         config = new LoggingSourceConfiguration(_logsLevel);
                     }
-                    if (_mode == PipeMode.Metrics)
+                    else if (_mode == PipeMode.Metrics)
                     {
-                        config = new MetricSourceConfiguration(_metricIntervalSeconds);
+                        config = new MetricSourceConfiguration(_metricIntervalSeconds, _counterFilter.GetProviders());
                     }
-                    if (_mode == PipeMode.GCDump)
+                    else if (_mode == PipeMode.GCDump)
                     {
                         config = new GCDumpSourceConfiguration();
                     }
-                    if (_mode == PipeMode.ProcessInfo)
+                    else if (_mode == PipeMode.ProcessInfo)
                     {
                         config = new SampleProfilerConfiguration();
                     }
+                    else if (_mode == PipeMode.Nettrace)
+                    {
+                        config = _userConfig;
+                    }
 
                     monitor = new DiagnosticsMonitor(config);
                     Stream sessionStream = await monitor.ProcessEvents(client, duration, token);
+
+                    if (_mode == PipeMode.Nettrace)
+                    {
+                        _sessionStarted.TrySetResult(true);
```

Comment:
If the completion source cannot be move to the successful completed state, we should check if the token cause it to be canceled so that we can cause the Process method to be canceled.